pandas/core/frame.py

```diff
@@ -5803,7 +5805,7 @@ def dropna(
         axis: Axis = 0,
         how: str = "any",
         thresh=None,
-        subset=None,
+        subset: Optional[Union[Hashable, Sequence[Hashable]]] = None,
```

Comment:
Could you type using new syntax | for Union and | None for Optional?

Reply:
I actually saw we have an internal type for this if I'm not mistaken:`IndexLabel`

Reply:
@simonjayhawkins would be the person to ask here

Reply:
@simonjayhawkins any chance you can have a look?